types/connector.d.ts

```diff
@@ -15,6 +15,7 @@ export interface Connector {
   disconnect(callback?: Callback): PromiseOrVoid; // Disconnect from the underlying system
   ping(callback?: Callback): PromiseOrVoid; // Ping the underlying system
   execute?(...args: any[]): Promise<any>;
+  [property: string]: any; // Other properties that vary by connectors
```

Comment:
At minimum please use `Function` instead of `any`.

Personally, I'd prefer to see a list of well-known properties declared as optional (see `execute` above).

Reply:
We cannot use function for the indexed properties because this type must support all well-defined properties too.